R/input-select.R

```diff
@@ -185,16 +185,58 @@ selectizeInput <- function(inputId, ..., options = NULL, width = NULL) {
 selectizeIt <- function(inputId, select, options, nonempty = FALSE) {
   res <- checkAsIs(options)
 
-  selectizeDep <- htmlDependency(
-    "selectize", "0.11.2", c(href = "shared/selectize"),
-    stylesheet = "css/selectize.bootstrap3.css",
-    head = format(tagList(
-      HTML('<!--[if lt IE 9]>'),
-      tags$script(src = 'shared/selectize/js/es5-shim.min.js'),
-      HTML('<![endif]-->'),
-      tags$script(src = 'shared/selectize/js/selectize.min.js')
-    ))
-  )
+  # Compile selectize Sass against a bootstraplib theme (if one exists)
+  selectizeDep <- if (useBsTheme()) {
+    scss <- system.file(package = "shiny", "www", "shared", "selectize", "scss")
+    is_bs3 <- "3" %in% bootstraplib::theme_version()
+    scss_file <- if (is_bs3) {
+      file.path(scss, "selectize.bootstrap3.scss")
+    } else {
+      file.path(scss, "selectize.bootstrap4.scss")
+    }
+    tmpdir <- tempfile("selectize-custom")
+    dir.create(tmpdir)
+    bootstraplib::bootstrap_sass(
+      output = file.path(tmpdir, "selectize-custom.css"),
+      rules = list(
+        # For some reason the selectize.bootstrap.scss hard codes these values
+        # assuming a normal white bg black fg theme...these are better default values
+        if (is_bs3) {
+          list(
+            "selectize-color-item" = "$gray-lighter !default;",
+            "selectize-color-item-border" = "$gray-light !default;",
+            "selectize-color-item-active-text" = "$white !default;",
+            "selectize-color-item-active-border" = "$gray-base !default;"
+          )
+        } else {
+          list(
+            "selectize-color-item" = "gray('100') !default;",
+            "selectize-color-item-border" = "gray('400') !default;",
+            "selectize-color-item-active-text" = "$white !default;",
+            "selectize-color-item-active-border" = "$black !default;"
+          )
```

Comment:
For some reason this isn't enough to get Bootswatch based dropdowns colored appropriately:

```r
library(bootstraplib)
bs_theme_new(bootswatch = "darkly")
bs_theme_preview()
```

Reply:
There are two problems here:

1. Bootswatch themes don't take the same approach as `bs_theme_base_colors()` (the latter sets the "base" colorscales, which effects all downstream color variables). I'm not sure whether it'll make sense for use to "patch" this in the sense that bootswatch color model matches `bs_theme_base_colors()`, but assuming we don't, this implies we can't use `gray()` at this point in the way I was hoping to.
2. Bootswatch themes like darkly and cyborg have two totally different color models for inputs (light bg, dark fg) and dropdowns (dark bg, light fg)....